### PR TITLE
HowToGuides/GettingStarted: require Git 2.x

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Trying to run `update-checkout` with Git 1.8.3.1 fails during rebasing. Git 2.38 works fine. Additionally, looks like Git 1.x requires committer name and email to be set during rebasing, which also causes errors. We should recommend Git 2.x explicitly to make sure new contributors don't get confused by this.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: docs/HowToGuides/GettingStarted.md [hunk 1]
```diff
@@ -45,12 +45,14 @@ toolchain as a one-off, there are a couple of differences:
    system. Experimental instructions for Windows are available under
    [Windows.md](/docs/Windows.md).
 2. Python 3: Several utility scripts are written in Python.
-3. Disk space:
+3. Git 2.x to check out the sources. We find that older versions of Git
+   raise errors when doing the full checkout.
+5. Disk space:
```
Comment (buttaface) [inline docs/HowToGuides/GettingStarted.md:50]: Should be 4.

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Review by AnthonyLatsis — APPROVED** [review]

**Pull request merged**